Open WorldCat.js

```diff
@@ -341,7 +340,17 @@ function fetchIDs(isbns, ids, callback) {
 				} else {
 					Z.debug("Could not extract OCLC ID for ISBN " + isbn);
 				}
-			} else {
+			}
+			//but sometimes we have single items
+			else if (getFirstContextObj(doc)) {
+				var canonicalURL = ZU.xpath(doc, '/html/head/link[@rel="canonical"][1]')[0];
```

Comment:
I would suggest:
```js
var canonicalURL = ZU.xpathText(doc, '/html/head/link[@rel="canonical"]/@href');
```
(It looks IMO not likely that a website has several canonical links.)